### PR TITLE
[FIX] rename `contrast_type` to `stat_type`

### LINKED ISSUE #4186: [BUG] inconsistency in the naming
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Operating system

- [X] Linux
- [ ] Mac
- [ ] Windows

### Operating system version

For example one of the following:
- Linux Ubuntu 22.04
- Mac OS Version 12 "monterey"
- Windows 11

### Python version

- [ ] 3.12
- [ ] 3.11
- [X] 3.10
- [ ] 3.9
- [ ] 3.8

### nilearn version

dev version

### Expected behavior

The same argument has different names whether called in the `compute_contrast` function of the corresponding method of the *GLM classes.
`contrast_type`  in the function, see  https://github.com/nilearn/nilearn/blob/cfc4c03f/nilearn/glm/contrasts.py#L66
`stat_type` in teh method, see https://github.com/nilearn/nilearn/blob/cfc4c03fe2533dfffd6a6f034f1d6404262e1ec1/nilearn/glm/first_level/first_level.py#L763C9-L763C9
We should  deprecate one of them. 
My imrpession is that `stat_type` is more accurate.

### Current behavior & error messages

This is what I got:

```bash
# Paste the error message here


```

### Steps and code to reproduce bug

```python
# Paste your code here


```

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4186

Changes proposed in this pull request:

- rename parameter `contrast_type` by `stat_type` in the function `compute_contrast`, see https://github.com/nilearn/nilearn/blob/cfc4c03f/nilearn/glm/contrasts.py#L66
- rename attribute `contrast_type` by `stat_type` the Contrast class
- apply similar change in downstream private functions

### CODE REVIEW

**Comment by bthirion** [comment]
We'll have to go through deprecation cycles for that one...

File: nilearn/glm/contrasts.py [hunk 1]
```diff
@@ -47,7 +48,10 @@ def expression_to_contrast_vector(expression, design_columns):
     return contrast_vector
 
 
-def compute_contrast(labels, regression_result, con_val, contrast_type=None):
+@rename_parameters(
+    replacement_params={"contrast_type": "stat_type"}, end_version="0.13.0"
+)
```
Comment (Remi-Gau) [inline nilearn/glm/contrasts.py:53]: This decorator should take care of the deprecation cycle.

Will add a test to make sure of it.

**Comment by Remi-Gau** [comment]
@bthirion 
for consistency I would also change the attribute contrast_type in the Contrast class.

https://github.com/nilearn/nilearn/blob/cfc4c03fe2533dfffd6a6f034f1d6404262e1ec1/nilearn/glm/contrasts.py#L178

What do you think?

**Comment by bthirion** [comment]
Indeed, the idea is to have a uniform naming across classes and functions.

File: nilearn/glm/first_level/first_level.py [hunk 2]
```diff
@@ -986,6 +986,7 @@ def first_level_from_bids(
     sub_labels : :obj:`list` of :obj:`str`, optional
         Specifies the subset of subject labels to model.
         If 'None', will model all subjects in the dataset.
+
```
Comment (Remi-Gau) [inline nilearn/glm/first_level/first_level.py:989]: This is unrelated but came across a directive that did not render in the doc.

https://nilearn.github.io/stable/modules/generated/nilearn.glm.first_level.first_level_from_bids.html#nilearn.glm.first_level.first_level_from_bids

image

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Pull request merged**